pipelines/google_ads/__init__.py

```diff
@@ -0,0 +1,141 @@
+from typing import Dict, List, Union
+import dlt
+from dlt.common.configuration.specs import GcpClientCredentialsWithDefault
+from pipelines.google_analytics.helpers.credentials import GoogleAnalyticsCredentialsOAuth
+from dlt.common.exceptions import MissingDependencyException
+from dlt.common.typing import DictStrAny, DictStrStr, TDataItem
+from dlt.common.pendulum import pendulum
+from pipelines.google_ads.helpers.data_processing import process_dimension, process_report, process_query
+try:
+    from google.ads.googleads.client import GoogleAdsClient
+except ImportError:
+    raise MissingDependencyException("Requests-OAuthlib", ["google-ads"])
+try:
+    from google.oauth2.credentials import Credentials
+except ImportError:
+    raise MissingDependencyException("Google OAuth Library", ["google-auth-oauthlib"])
+
+
+DIMENSION_TABLES = ["accounts", "ad_group", "ad_group_ad", "ad_group_ad_label", "ad_group_label", "campaign_label", "click_view", "customer", "keyword_view", "geographic_view"]
+REPORT_TABLES = []
+
+
+@dlt.source(max_table_nesting=2)
+def google_ads_query(credentials: Union[GoogleAnalyticsCredentialsOAuth, GcpClientCredentialsWithDefault] = dlt.secrets.value,
+                     dev_token: str = dlt.secrets.value,
+                     query_list: List[DictStrAny] = dlt.config.value):
+    """
+    This source reads custom queries from the API.
+    :param credentials:
+    :param dev_token:
+    :param query_list:
+    :return:
+    """
+
+    # generate access token for credentials if we are using OAuth2.0
+    if isinstance(credentials, GoogleAnalyticsCredentialsOAuth):
+        credentials.auth("https://www.googleapis.com/auth/analytics.readonly")
+        credentials = {
```

Comment:
you can just use `to_native_representation` if you import from our specs (I hope - if not then tell me)

Reply:
or just use `to_google_credentials` that will return google credentials object directly

Reply:
we need a dict here to inject the developer key, somehow google ads works a bit different. we can discuss